janitor/functions.py

```diff
@@ -2810,42 +2810,81 @@ def dropnotnull(df: pd.DataFrame, column_name) -> pd.DataFrame:
 
 @pf.register_dataframe_method
 @deprecated_alias(column="column_name")
-def find_replace(df: pd.DataFrame, column_name, mapper: Dict) -> pd.DataFrame:
+def find_replace(
+    df: pd.DataFrame, column_name, mapper: Dict, match: str = "exact"
+) -> pd.DataFrame:
     """
     Perform a find-and-replace action on a column of data.
 
     This method mutates the original DataFrame.
 
-    For example, let's say we have a column for which we want to replace all
-    of the values 'a' with 1, 'b' with 2, 'c' with 3. We would use the
-    following function call:
+    Depending on use cases, users can choose either exact, full value matching
+    or regular-expression-based fuzzy matching (substring matching is allowed
+    in the latter case). For strings, the matching is always case sensitive.
+
+    For instance, given a dataframe containing orders at a coffee shop:
 
     .. code-block:: python
 
-        df = (
-            pd.DataFrame(...)
-            .find_replace('column_name', {'a': 1, 'b': 2, 'c': 3})
+        df = pd.DataFrame({
+            'customer': ['Mary', 'Tom', 'Lila'],
```

Comment:
I like the new example! Thanks for this, @shandou :smile:

Reply:
Just added related info to change log and had the realization that I had not properly synched my fork this whole time 😳... I thought `git fetch upstream` was all that was needed to stay in sync with the mothership but I was missing a lot of changes--including not having the file `CHANGELOG.rst` in my fork, and my environment yaml was also outdated. Just checked github docs on [how to sync forks](https://help.github.com/en/articles/syncing-a-fork) properly and merged `upstream/dev` to my fork's local branch to resolve this issue.

Speaking of which, it might also be helpful to add some related guidance to the docs (I am assuming I may not be the only person who got confused on how to stay in sync 😅)

Reply:
> Speaking of which, it might also be helpful to add some related guidance to the docs (I am assuming I may not be the only person who got confused on how to stay in sync 😅)

Yes! I agree - this is something that confuses even myself at times.

I will raise an issue.